src/shared/seccomp-util.c

```diff
@@ -699,6 +700,50 @@ static int seccomp_add_syscall_filter_set(
         return 0;
 }
 
+int seccomp_add_syscall_filter_set_whitelist(
+                scmp_filter_ctx seccomp,
+                uint32_t default_action,
+                const SyscallFilterSet *set,
+                char **syscall_whitelist,
+                uint32_t action) {
+
+        const char *sys;
+        int r;
+
+        assert(seccomp);
+        assert(set);
+
+        NULSTR_FOREACH(sys, set->value) {
+                int id;
+
+                /* We can also whitelist entire filter sets */
+                if (sys[0] == '@' && !strv_contains(syscall_whitelist, sys)) {
```

Comment:
both if branches check `!strv_contains(…)`. Hence you can pull it out of the if check and instead continue early, no? That would remove redundancy and also be more readable... i.e. something like this at the beginning fo the loop:

```c
if (strv_containers(syscall_whitelist, sys))
         continue;
```